pkg/fission-cli/cmd/spec/apply.go

```diff
@@ -486,16 +489,18 @@ func localArchiveFromSpec(specDir string, aus *spectypes.ArchiveUploadSpec) (*fv
 	}
 
 	if len(files) > 1 || !isSingleFile {
-		// zip up the file list
-		archiveFile, err := os.CreateTemp("", fmt.Sprintf("fission-archive-%v", aus.Name))
+		// zip up the file list.Adding .zip because archive function expects that.
+		archiveFile, err := os.CreateTemp("", fmt.Sprintf("fission-archive-%v-*.zip", aus.Name))
 		if err != nil {
 			return nil, err
 		}
 		archiveFileName = archiveFile.Name()
-		err = archiver.Zip.Make(archiveFileName, files)
+		archiver.DefaultZip.OverwriteExisting = true
```

Comment:
Let's add a comment here, why we need to set it `true` and reset to `false` afterwards.

Reply:
I do see a issue here though, if we have a concurrent calls happening to `Archive` and we setting `archiver.DefaultZip.OverwriteExisting` to `true`